### ISSUE TITLE
Increase default value for max_iter in ShapeletModel

### ISSUE THREAD

**Issue by johannfaouzi** [rtavenar/tslearn#179]
**Describe the bug**
With the current value `max_iter=100`, the model is far from converging. It could be useful to change the default values (either increasing the maximum number of iterations or the learning rate) so that the default behavior would lead to better results.

**To Reproduce**
```python
>>> from tslearn.shapelets import ShapeletModel
>>> from tslearn.datasets import UCR_UEA_datasets

>>> X_train, y_train, X_test, y_test = UCR_UEA_datasets().load_dataset('Coffee')
>>> clf = ShapeletModel(random_state=42)
>>> clf.fit(X_train, y_train)
>>> clf.score(X_test, y_test)
0.5
```

```python
>>> clf = ShapeletModel(max_iter=10000, random_state=42)
>>> clf.fit(X_train, y_train)
>>> clf.score(X_test, y_test)
0.9642857142857143
```

**Side Note**
It could be useful to benchmark the performance of the implementations in this package to the literature (results in original papers, results from the timeseriesclassification website, etc.).

**Comment by rtavenar** [comment]
Hi @johannfaouzi 

This would indeed make sense. One thing we should keep in mind however is that tests should be able to run in a reasonable amount of time. But if I'm correct, number of epochs is enforced at test time [there](https://github.com/rtavenar/tslearn/blob/master/tslearn/tests/test_estimators.py#L168-L169).

Concerning your side note, this is a real problem (as far as I know) with Shapelets: results vary a lot from one implementation to the other (cf for example results obtained by Tony Bagnall reported on timeseriesclassification.com and those from the original paper -- that are also different if I remember correctly from those reported on the paper's web page).

**Comment by rtavenar** [comment]
I have one naive question regarding this change. Ideally, we would like to warn any user that does not set the `max_iter` attribute that its default value has changed between `tslearn` versions. But is there a way to detect that properly, or are we forced to do something like 

```python
if max_iter == NEW_DEFAULT_VALUE:
    warnings.warn(...)
```

?

Anyway I could do the change before releasing v0.3.

**Comment by johannfaouzi** [comment]
> This would indeed make sense. One thing we should keep in mind however is that tests should be able to run in a reasonable amount of time. But if I'm correct, number of epochs is enforced at test time [there](https://github.com/rtavenar/tslearn/blob/master/tslearn/tests/test_estimators.py#L168-L169).

Yes, the goal of most tests is not to check if convergence occurred and if the predictive performance is great (these tests should not be part of the CI imho, they take way too long, but could be checked elsewhere).

> I have one naive question regarding this change. Ideally, we would like to warn any user that does not set the `max_iter` attribute that its default value has changed between `tslearn` versions. But is there a way to detect that properly, or are we forced to do something like
> 
> ```python
> if max_iter == NEW_DEFAULT_VALUE:
>     warnings.warn(...)
> ```
> 
> ?

I think that we have to do something like this. This is usually what is done in scikit-learn. Here is an example for the change of `gamma` in SVC:
https://github.com/scikit-learn/scikit-learn/blob/ee328faa3601b40944ad43e28bce71860d39f2de/sklearn/svm/base.py#L182-L193

**Comment by rtavenar** [comment]
This will appear in v0.3, and should be properly documented. Thanks @johannfaouzi for the input!
